streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -178,6 +180,20 @@ public void testProcessOrder() throws Exception {
 
     }
 
+    @SuppressWarnings("unchecked")
+    @Test
+    public void shouldAddSeparateTimestampExtractorsPerEachSource() throws Exception {
+        StreamsConfig config = createConfig(baseDir);
+        final TopologyBuilder builder = new TopologyBuilder();
+        builder.addSource(new MockTimestampExtractor(), source1.name(), topic1[0]);
+        builder.addSource(new MockTimestampExtractor2(), source2.name(), topic2[0]);
```

Comment:
The test is fine -- but what's the value in testing overwrite the default extractor two times.